exercises/queen-attack/src/example/java/QueenAttackCalculator.java

```diff
@@ -0,0 +1,86 @@
+import java.util.Arrays;
+
+public final class QueenAttackCalculator {
+
+    private final int[] queen1Coordinates;
+
+    private final int[] queen2Coordinates;
+
+    public QueenAttackCalculator(final int[] queen1Coordinates, final int[] queen2Coordinates)
```

Comment:
subject `Coordinates`:
It is absolutely "in bounds" to include other bits in the starter code (i.e. `main` sourceset).  There are examples of this in other exercises ([`Allergen`](https://github.com/exercism/xjava/blob/master/exercises/allergies/src/main/java/Allergen.java) for one).

The way I think about it, we are defining the API for the exercising programmer.  We are free to define as little as an interface and as much as a whole class as we see appropriate.